app/src/main/java/es/situm/gettingstarted/drawbuilding/DrawBuildingActivity.java

```diff
@@ -83,23 +108,71 @@ void drawBuilding(Bitmap bitmap) {
     }
 
-    void fetchFirstFloorImage(Building building, Callback callback) {
+    void fetchFloorsImages(Building building, Callback callback) {
         SitumSdk.communicationManager().fetchFloorsFromBuilding(building.getIdentifier(), new Handler<Collection<Floor>>() {
             @Override
             public void onSuccess(Collection<Floor> floorsCollection) {
                 List<Floor> floors = new ArrayList<>(floorsCollection);
-                Floor firstFloor = floors.get(0);
-                SitumSdk.communicationManager().fetchMapFromFloor(firstFloor, new Handler<Bitmap>() {
+
+                // First instance a new LinkedHashMap to store all the floors in the same order they came [FIFO]
+                buildingFloors = new LinkedHashMap<>();
+                for(int i=floors.size()-1;i>=0;i--){
+                    Floor floor = floors.get(i);
+                    buildingFloors.put(floor.getIdentifier(), floor);
+                }
+
+                // Secondly, if the building has more than 3 floors, display the scroll marks to indicate it
+                if(buildingFloors.size() > 3){
+                    selectorMarkTop.setVisibility(View.VISIBLE);
+                    selectorMarkBottom.setVisibility(View.VISIBLE);
+                }
+
+                // Secondly instance the adapter of the selector
+                floorAdapter = new FloorAdapter(new ArrayList<>(buildingFloors.values()), floor -> {
+
+                    SitumSdk.communicationManager().fetchMapFromFloor(floor, new Handler<Bitmap>() {
+                        @Override
+                        public void onSuccess(Bitmap bitmap) {
+                            callback.onSuccess(bitmap);
+                            floorAdapter.select(floor);
+                            int selectedFloorIndex = floorAdapter.getSelectedFloorIndex();
+                            //When the selected item naturally occupies one of the recycler's hidden positions, it
+                            //needs to be moved (scrolled) to make it visible
+                            LinearLayoutManager llm = (LinearLayoutManager) selector.getLayoutManager();
+                            if (selectedFloorIndex < llm.findFirstCompletelyVisibleItemPosition() &&
+                                    selectedFloorIndex > llm.findLastCompletelyVisibleItemPosition()) {
+                                selector.scrollToPosition(selectedFloorIndex);
+                            }
+                        }
+
+                        @Override
+                        public void onFailure(Error error) {
+                            callback.onError(error);
+                        }
+                    });
+
+                });
+
+                Floor defaultFloor = floors.get(0);
+
+                // The next method selects by default the first item of the List<>
+                SitumSdk.communicationManager().fetchMapFromFloor(defaultFloor, new Handler<Bitmap>() {
```

Comment:
Esta segunda chamada ao método `fetchMapFromFloor` deberías poder evitala.
Estás adaptando o que había antes, que pescaba unha planta do edificio e a pintaba. Eso é natural, pero neste caso o refactor creo que pide un pouquiño máis.
Para min o natural sería:
1. Partir dun conxunto baleiro de datos e inicializar todo (adapters, listeners, o que sexa): o código debe saber comportarse aínda que haxa 0 plantas.
2. Pedir as plantas do edificio. Unha vez as tes, actualizas o teu _underlying dataset_.
3. Cando xa tes datos, seleccionas unha planta por defecto, utilizando os mecanismos que xa tes instanciados, non unha nova chamada a `fetchMapFloor`.